### ISSUE TITLE
Clipboard content modified on copy/paste : pasting binary or Japanese characters on client

### ISSUE THREAD

**Issue by LeadManPL** [deskflow/deskflow#6582]
### Operating Systems ###
* Server: Ubuntu 18.04.3 LTS
* Client: Windows 7 

### Synergy Version ###
```
root@patben-amd:~/.synergy# LANG=C;apt-cache policy synergy 
synergy:
  Installed: 1.10.3.stable~b24+ca35737a
  Candidate: 1.10.3.stable~b24+ca35737a
  Version table:
 *** 1.10.3.stable~b24+ca35737a 100
        100 /var/lib/dpkg/status
     1.8.8-stable+dfsg.1-1build1 500
        500 http://pl.archive.ubuntu.com/ubuntu bionic/universe amd64 Packages
```
![client_windows_version](https://user-images.githubusercontent.com/13769051/67485972-66508f80-f66b-11e9-9862-6682b66ff5b9.PNG)

### Steps to Reproduce ###
**Reproducible: often**

1. Set server & client to debug2 mode
1. Start Evolution MUA on linux server
1. find an email with some  URL links in body
1. right click on link
1. select "Copy link address"
1. Move mouse cursor to windows client
1. Press ^v at any editor

* Expected: Awesome things to happen, like actually pasting an URL link from server.
* Actual: Not so awesome things happened: some Japanese or Chinese text is pasted instead.

### Log URLs ###
[This instruction](https://symless.com/help/s1-troubleshooting/s1-sending-logs) provided in bug report template leads to non existing web page.

According to [GitHub policy](https://help.github.com/en/github/managing-your-work-on-github/file-attachments-on-issues-and-pull-requests), attaching log files to issues is acceptable, so log attached: [synergy_logs.zip](https://github.com/symless/synergy-core/files/3767766/synergy_logs.zip)

### Extra Info ###

* Attach any screenshots
  * This error occured ~15:03:03 in log file
![paste_error](https://user-images.githubusercontent.com/13769051/67486644-b1b76d80-f66c-11e9-88cf-dd092f27f79b.PNG)
  * This error occured ~15:22:50 in log file
![paste_error_2](https://user-images.githubusercontent.com/13769051/67488290-dc56f580-f66f-11e9-9bdd-451a83d5859c.PNG)
* What maybe causing it?
  * I have no clue, but observed only with Evolution MUA
* Are there any workarounds?
  * Yes, type copied text manulally :/
* What other apps are you running?
  * Seriously?
* Does it stop you using Synergy entirely?
  * No
* Did restarting Synergy or the computer help?
  * No
* Is there anything else you can think of
  * I think I'll grab some ice cream ;)

**Comment by LeadManPL** [comment]
I was unable to add label :
/~v1.x

**Comment by asihera** [comment]
I can confirm this is a problem with synergy 1.8.8 (protocol version 1.6), between two vanilla Ubuntu 18.10 installations with the regional language set to UTF-8 on both sides.

Copying any kind of foreign text from one machine to the other just results in a line of question marks being inserted at the destination. It doesn't seem to matter what the source or destination applications are (I confirmed the problem between various combinations of GNU Terminal, GNU Edit, Firefox, and Thunderbird).

**Comment by khamer** [comment]
I see this on 1.10 too. I think there's something getting messed up, as Synergy is trying to convert the URL to rich text or something - I also sometimes see generated, malformed HTML snippets get pasted instead that look like this:
```
<�a href="https://github.com/symless/synergy-core/issues/6582">https://github.com/symless/synergy-core/issues/6582<�/a>
```
It might be a separate issue, but I believe they're related.

**Comment by LeadManPL** [comment]
Hi @Jnewbon , why this was closed? I still get this error. Should I turn on watch settings on #6809 ?

**Comment by Jnewbon** [comment]
@LeadManPL Yeah this was merged into master in this PR #6810,

We haven't done a release yet, did you build from source? if you have then we will need to reopen, but the code has not reached our release builds yet. which will probably be 1.13.0.

**Comment by LeadManPL** [comment]
I use deb packages, so I'll wait for release. Thanks.

**Comment by andreacndn** [comment]
It's fixed for macOS, it will be fixed for Windows-Ubuntu in 1.14.1

**Comment by LeadManPL** [comment]
Any ETA for 1.14.1 DEB package?

**Comment by andreacndn** [comment]
We are today on 1.14 release candidate, it will be a few weeks for 1.14.1

**Comment by dneprqa** [comment]
Fixed in 1.14.1

**Comment by LeadManPL** [comment]
Seems like 1.14.1 fixed this issue, I can't reproduce it anymore.
Thanks! :)

**Comment by nPHYN1T3** [comment]
This is still very much a problem on deskflow 1.17.0.163 in 2024. Clipboard has tons of issues...sadly the same issues from like 10 years ago heh. Synergy, Barrier, Input Leap, Deskflow all still carry the same issues forward.

**Comment by nbolton** [comment]
Ok, will re-open. Sorry to hear you're still having issues. Could you give some examples of what you're unable to copy and paste?

**Comment by sithlord48** [comment]
> This is still very much a problem on deskflow 1.17.0.163 in 2024. Clipboard has tons of issues...sadly the same issues from like 10 years ago heh. Synergy, Barrier, Input Leap, Deskflow all still carry the same issues forward. & ~E_NOTICE) becomes ☠縠彅低䥔䕃

Yes i think alot of this needs to be rewritten as i would suspect alot comes from a time before unicode was common

**Comment by nPHYN1T3** [comment]
Examples of things I can't copy paste? Honestly everything. That said this thread is about the whole you copy the word "book" and get some emoji Kanji translation. It seems most times that happens is copying something from the web. So URL's or text from links is most often (but not limited to) this translation.

As for copy paste as a whole it's a frustrating nightmare as a whole. I am constantly fighting to get anything to copy or paste. I had thought perhaps my clipboard managers were somehow causing extra issues but upon killing them it's the same. If you copy anything you get the dice roll when you paste it will be Kanji or symbols but more so you copying something from one machine and trying to move to another results in the cursor being locked to the first machine for a moment. Then even when the second machine picks up (and reports the clipboard has been updated) often times when you paste you get nothing.

I could see this behavior if someone was copy and pasting images or massive volumes of text but I'm talking a command, a sentence, a word and everything gets all "Whoa slow down 'cat' is a lot to process, give me a minute!" then when you paste...nothing. Often if you paste via right click context the paste will be grayed out because the buffer has been cleared rather than synced / filled.

The only thing the logs ever says/shows that looks/sounds like an issue is stuff like "pdate of clipboard 1 (missequenced)" or/and "ERROR: corrupted clipboard data, expected size=4 actual size=8." The rub is that when I get that in the logs it usually works heh.

**Comment by nbolton** [comment]
Interesting. I rarely if ever have copy/paste issues (except for on Wayland where it isn't implemented yet). That said I don't use clipboard manager apps or Kanji. Do you think those two things are potentially the cause? Have you tried isolating the issue by disabling the clipboard managers or avoiding Kanji?

**Comment by nPHYN1T3** [comment]
Yes all machines are running X (Wayland breaks MultiGPU).

**Comment by sithlord48** [comment]
Are the machines using the same locale ? 
What De are you using  any other relevant stack info ?

**Comment by nPHYN1T3** [comment]
No DE, all machines are the same locale.

**Comment by nPHYN1T3** [comment]
Again, I've dealt with this for years on many machines with only a single XScreen too, it doesn't change anything.

**Comment by sithlord48** [comment]
I've used the forks for years and i've never seen this .

**Comment by sithlord48** [comment]
Im not sure about how the internal stuff is set up ill take  a look and see if i see anything that stands out.. may i ask what locale your using for your systems

**Comment by nPHYN1T3** [comment]
heh I have. Even when running Windows to Windows...man thinking about this crap makes me feel old. Back before Synergy went all commercial I'd have this just not as frequently.

**Comment by nPHYN1T3** [comment]
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=

**Comment by sithlord48** [comment]
Well that is odd as I am also on arch w/ the same locale and if i switch to a non wayland session i have no issues with the clipboard between machines  (i have linux / mac clients no windows ones). I guess the main difference is i have a DE in this case (KDE) and you don't.

**Comment by nPHYN1T3** [comment]
Unfortunately I hunt a lot of ghosts where there is some "kitchen sink" dependency that's not declared and just kinda assumed to be with a DE that I don't have. That said, this has been an issue even way back to the days when I had a few Windows machines, it just wasn't as bad. I think it's extra grumpy on Lin because every machine has 2 clipboards which may or may not play nice.

**Comment by sithlord48** [comment]
yeah i use the kde clipboard manager

**Comment by nbolton** [comment]
> heh I have. Even when running Windows to Windows...man thinking about this crap makes me feel old. Back before Synergy went all commercial I'd have this just not as frequently.

It could be that some change in your OS environment coincides with some release of Synergy and the two events may not be related... have you tried running on a fresh install without any other software installed? For example, you could try running virtual machines. When was the last time you reinstalled your OS from fresh?

Speaking from personal experience, over the years the clipboard sharing feature in Synergy has become more stable for me. The Synergy team (paid employees) invested at least several hundred hours in improving the stability of the clipboard sharing and other features. It is possible though that a bug fix for some has made it worse for others.

**Comment by nPHYN1T3** [comment]
@nbolton I'm not sure if you're trolling (:P), not reading or insomnia has made my communication deranged but...I don't run Windows. ;p As I said, I had the issue Win 2 Win BACK before Synergy went commercial, so like 20 years ago.

**Comment by nPHYN1T3** [comment]
y0 y0 I wanted to toss something in here I've noticed that may or may not be related to this issue. When I'm trying to copy paste most the time it just doesn't work at all. Then when it "does" it's a dice roll if I get what I copied or the Kanji, emoji, random character scenario. It seems like this might be an input duration issue.

I'm just spitballing but if I do a normal ctrl + C it often fails or mutilates the text. If I slow WAY down it seems to work more. So it seems like deskflow isn't polling fast enough. More so when I copy then leave the one screen fast it's like it stops trying to complete the previous task leaving the encoding of text garbled. Then when you paste you get the half baked incomplete data that comes out as garbage.

I've also noticed when highlighting on the client machine deskflow likes to toss in what seems like extra clicks. So click and drag over text and then it will auto clear like I clicked again when in reality all I did was mouse button up. So something is screwy in general with input and I'd guess this is all related.

**Comment by nPHYN1T3** [comment]
I was trying to catch another bug/issue by setting my logs to debug 1 and 2 and with no input saw my logs being flooded with
[2024-12-02T15:18:18] DEBUG1: clipboard: sending failure to 0x02200001,336,0
[2024-12-02T15:18:18] DEBUG1: request for clipboard 1, target UTF8_STRING (336) by 0x02200001 (property=GREENCLIP (548))
[2024-12-02T15:18:18] DEBUG1: success
[2024-12-02T15:18:18] DEBUG1: clipboard: setting property on 0x02200001,336,548
[2024-12-02T15:18:18] DEBUG1: clipboard: sending notify to 0x02200001,336,548
[2024-12-02T15:18:19] DEBUG1: request for clipboard 448, target UTF8_STRING (336) by 0x02200001 (property=GREENCLIP (548))
[2024-12-02T15:18:19] DEBUG1: failed
[2024-12-02T15:18:19] DEBUG1: clipboard: sending failure to 0x02200001,336,0
[2024-12-02T15:18:19] DEBUG1: request for clipboard 1, target UTF8_STRING (336) by 0x02200001 (property=GREENCLIP (548))
[2024-12-02T15:18:19] DEBUG1: success
[2024-12-02T15:18:19] DEBUG1: clipboard: setting property on 0x02200001,336,548
[2024-12-02T15:18:19] DEBUG1: clipboard: sending notify to 0x02200001,336,548
[2024-12-02T15:18:20] DEBUG1: request for clipboard 448, target UTF8_STRING (336) by 0x02200001 (property=GREENCLIP (548))
[2024-12-02T15:18:20] DEBUG1: failed

**Comment by nPHYN1T3** [comment]
So I just tried Deskflow: 1.19.0.8 (10c50c97)
Qt: 6.8.2
System: Arch Linux
Session: CTWM (x11)

It immediately glitched out (cursor was vanishing) and all the logs would show was each screen was constantly updating the clipboard over and over when nothing new had been added to it. However upon start something was put in the clip board "�" so immediate glyph/kanji/emoji/garbage issues.

```
2025-02-03T16:50:11] INFO: screen "pillar" grabbed clipboard 1 from "fsrv1"
[2025-02-03T16:50:11] INFO: screen "fsrv0" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:11] INFO: screen "pillar" grabbed clipboard 1 from "fsrv0"
[2025-02-03T16:50:11] INFO: screen "fsrv1" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:11] INFO: screen "fsrv1" grabbed clipboard 0 from "fsrv2"
[2025-02-03T16:50:11] INFO: screen "fsrv2" grabbed clipboard 1 from "fsrv1"
[2025-02-03T16:50:11] INFO: screen "pillar" grabbed clipboard 1 from "fsrv2"
[2025-02-03T16:50:11] INFO: screen "fsrv2" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:11] INFO: screen "pillar" grabbed clipboard 1 from "fsrv2"
[2025-02-03T16:50:11] INFO: screen "fsrv0" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:11] INFO: screen "fsrv0" grabbed clipboard 0 from "fsrv1"
[2025-02-03T16:50:11] INFO: screen "pillar" grabbed clipboard 1 from "fsrv0"
[2025-02-03T16:50:11] INFO: screen "fsrv1" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:11] INFO: screen "pillar" grabbed clipboard 1 from "fsrv1"
[2025-02-03T16:50:11] INFO: screen "fsrv2" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:11] INFO: screen "pillar" grabbed clipboard 1 from "fsrv2"
[2025-02-03T16:50:11] INFO: screen "fsrv1" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:12] INFO: screen "fsrv2" grabbed clipboard 0 from "fsrv0"
[2025-02-03T16:50:12] INFO: screen "fsrv1" grabbed clipboard 0 from "fsrv2"
[2025-02-03T16:50:14] INFO: screen "pillar" grabbed clipboard 1 from "fsrv1"
[2025-02-03T16:50:14] INFO: screen "fsrv0" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:14] INFO: screen "pillar" grabbed clipboard 1 from "fsrv0"
[2025-02-03T16:50:14] INFO: screen "fsrv1" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:14] INFO: screen "fsrv2" grabbed clipboard 1 from "fsrv1"
[2025-02-03T16:50:14] INFO: screen "fsrv0" grabbed clipboard 0 from "fsrv1"
[2025-02-03T16:50:14] INFO: screen "pillar" grabbed clipboard 1 from "fsrv2"
[2025-02-03T16:50:14] INFO: screen "fsrv2" grabbed clipboard 0 from "fsrv0"
[2025-02-03T16:50:14] INFO: screen "fsrv0" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:14] INFO: screen "pillar" grabbed clipboard 1 from "fsrv0"
[2025-02-03T16:50:14] INFO: screen "fsrv2" grabbed clipboard 0 from "fsrv2"
[2025-02-03T16:50:14] INFO: screen "fsrv0" grabbed clipboard 0 from "fsrv2"
[2025-02-03T16:50:16] INFO: screen "fsrv1" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:16] INFO: screen "fsrv0" grabbed clipboard 1 from "fsrv1"
[2025-02-03T16:50:16] INFO: screen "fsrv0" grabbed clipboard 0 from "fsrv0"
[2025-02-03T16:50:16] INFO: screen "fsrv1" grabbed clipboard 0 from "fsrv0"
[2025-02-03T16:50:17] INFO: screen "fsrv2" grabbed clipboard 1 from "fsrv0"
[2025-02-03T16:50:17] INFO: screen "pillar" grabbed clipboard 1 from "fsrv2"
[2025-02-03T16:50:17] INFO: screen "fsrv2" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:17] INFO: screen "pillar" grabbed clipboard 1 from "fsrv2"
[2025-02-03T16:50:17] INFO: screen "fsrv1" updated clipboard 0
[2025-02-03T16:50:17] INFO: screen "fsrv1" updated clipboard 0
[2025-02-03T16:50:17] INFO: screen "fsrv0" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:17] INFO: screen "pillar" grabbed clipboard 1 from "fsrv0"
[2025-02-03T16:50:17] INFO: screen "fsrv0" grabbed clipboard 0 from "fsrv1"
[2025-02-03T16:50:17] INFO: screen "fsrv1" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:17] INFO: screen "fsrv2" grabbed clipboard 0 from "fsrv0"
[2025-02-03T16:50:17] INFO: screen "pillar" grabbed clipboard 1 from "fsrv1"
[2025-02-03T16:50:17] INFO: screen "fsrv1" grabbed clipboard 0 from "fsrv2"
[2025-02-03T16:50:17] INFO: screen "fsrv0" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:17] INFO: screen "pillar" grabbed clipboard 1 from "fsrv0"
[2025-02-03T16:50:17] INFO: screen "fsrv0" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:17] INFO: screen "pillar" grabbed clipboard 1 from "fsrv0"
[2025-02-03T16:50:17] INFO: screen "fsrv2" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:17] INFO: screen "fsrv1" grabbed clipboard 1 from "fsrv2"
[2025-02-03T16:50:17] INFO: screen "fsrv1" updated clipboard 0
[2025-02-03T16:50:17] INFO: screen "pillar" grabbed clipboard 1 from "fsrv1"
[2025-02-03T16:50:17] INFO: screen "fsrv0" grabbed clipboard 0 from "fsrv1"
[2025-02-03T16:50:17] INFO: screen "fsrv2" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:17] INFO: screen "pillar" grabbed clipboard 1 from "fsrv2"
[2025-02-03T16:50:17] INFO: screen "fsrv1" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:17] INFO: screen "pillar" grabbed clipboard 1 from "fsrv1"
[2025-02-03T16:50:17] INFO: screen "fsrv2" grabbed clipboard 0 from "fsrv0"
[2025-02-03T16:50:17] INFO: screen "fsrv1" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:17] INFO: screen "pillar" grabbed clipboard 1 from "fsrv1"
[2025-02-03T16:50:17] INFO: screen "fsrv0" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:17] INFO: screen "pillar" grabbed clipboard 1 from "fsrv0"
[2025-02-03T16:50:17] INFO: screen "fsrv0" grabbed clipboard 1 from "pillar"
[2025-02-03T16:50:18] INFO: screen "fsrv0" grabbed clipboard 0 from "fsrv2"
[2025-02-03T16:50:18] INFO: screen "fsrv1" grabbed clipboard 0 from "fsrv0"
[2025-02-03T16:50:19] INFO: screen "pillar" grabbed clipboard 1 from "fsrv0"
[2025-02-03T16:50:19] INFO: screen "pillar" grabbed clipboard 0 from "fsrv1"
```

I had been waiting on Synergy 1.17.1? to test but it seems the release is still MIA. I'm guessing the new Synergy build won't be any better...if it ever gets released. Forgive my pessimism but none of the bugs I've reported have been addressed. SSL is still broken in this Deskflow build and it looks like all the other bugs are in full swing as well. So far my hardware KM switch with x2x isn't flawless but it's light years ahead of what I'm seeing with Synergy/DeskFlow. With x2x copy paste works 100% of the time, my mouse never vanishes or stutters and everything over ssh.

**Comment by nbolton** [comment]
> I had been waiting on Synergy 1.17.1? to test but it seems the release is still MIA. I'm guessing the new Synergy build won't be any better...if it ever gets released. Forgive my pessimism but none of the bugs I've reported have been addressed. SSL is still broken in this Deskflow build and it looks like all the other bugs are in full swing as well. So far my hardware KM switch with x2x isn't flawless but it's light years ahead of what I'm seeing with Synergy/DeskFlow. With x2x copy paste works 100% of the time, my mouse never vanishes or stutters and everything over ssh.

Thanks for your feedback and for sticking with us. I understand your frustration with the bugs, especially the SSL/TLS bugs and the problems with copy/paste and mouse behavior.  We're releasing Synergy v1.17.1 today.  However, I want to be transparent: since Synergy v1.17.1 is based on Deskflow v1.17, and the SSL issue is still present in the latest Deskflow v1.19, it's unlikely that this Synergy release will resolve it.  We are actively working on SSL/TLS bugs in Deskflow, and the fix will be included in a future Deskflow and subsequent Synergy release.  Thanks for your patience.  Perhaps you could link to the specific SSL/TLS bugs you're seeing or raise them if they don't exist already.

**Comment by nPHYN1T3** [comment]
I've already reported the SSL issues here for Deskflow (https://github.com/deskflow/deskflow/issues/7883) I've reached out to Synergy support many times, several of which you replied to.

As for "sticking with us" I haven't, I bought hardware.

**Comment by nbolton** [comment]
> I've already reported the SSL issues here for Deskflow ([#7883](https://github.com/deskflow/deskflow/issues/7883)) I've reached out to Synergy support many times, several of which you replied to.

Thanks. I read dozens of issue comments per day and don't have a perfect memory of your comments. Thanks for participating in the discussion.

> As for "sticking with us" I haven't, I bought hardware.

That's unfortunate. We will try to fix the bugs you reported and I look forward to welcoming you back some day.

**Comment by nPHYN1T3** [comment]
There is nothing unfortunate about it. I can use my machines reliably now.

I'd love to be welcomed back but given this particular bug has been around since the early 2000's I'll be in the "Press X to doubt"  camp until proven otherwise. Likewise none of the other bugs I've reported this year have been addressed thus far. You have all my reports, all my logs, kernel dumps, even how my systems are set up. I'll check these issues from time to time to see if they have been closed as fixed at which time I may rebuild and test.

XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
(Does the joke land? I'll never know...)
